sdk/eventhub/event-hubs/test/eventHubConsumerClient.spec.ts

```diff
@@ -225,6 +225,200 @@ describe("EventHubConsumerClient", () => {
+                ownerLevel: 1,
+                maxBatchSize: 1,
+                maxWaitTimeInSeconds: 1
+              });
+            } else {
+              // stop this subscription, we know close was called so we've restarted
+              await subscription1.close();
+            }
+          },
+          async processClose() {
+            handlerCalls.close++;
+          },
+          async processInitialize() {
+            handlerCalls.initialize++;
+          }
+        };
+        const subscriptionHandlers2: SubscriptionEventHandlers = {
+          async processError() {},
+          async processEvents() {
+            // stop this subscription since it already should have forced the 1st subscription to have an error.
+            await subscription2!.close();
+          }
+        };
+        let subscription2: Subscription | undefined;
+        const subscription1 = consumerClient1.subscribe(partitionId, subscriptionHandlers1, {
+          maxBatchSize: 1,
+          maxWaitTimeInSeconds: 1
+        });
+
+        await loopUntil({
+          maxTimes: 10,
+          name: "Wait for subscription1 to recover",
+          timeBetweenRunsMs: 5000,
+          async until() {
+            return !subscription1.isRunning && !subscription2!.isRunning;
+          }
+        });
+
+        handlerCalls.initialize.should.be.greaterThan(1);
+        handlerCalls.close.should.be.greaterThan(1);
+      });
+
+      it("when subscribed to multiple partitions", async function(): Promise<void> {
+        const consumerClient1 = new EventHubConsumerClient(
+          EventHubConsumerClient.defaultConsumerGroupName,
+          service.connectionString,
+          service.path
+        );
+        const consumerClient2 = new EventHubConsumerClient(
+          EventHubConsumerClient.defaultConsumerGroupName,
+          service.connectionString,
+          service.path
+        );
+
+        clients.push(consumerClient1, consumerClient2);
+
+        const partitionIds = await consumerClient1.getPartitionIds();
+
+        const partitionHandlerCalls: {
+          [partitionId: string]: {
+            initialize: number;
+            processEvents: number;
+            close: number;
+          };
+        } = {};
+
+        // keep track of the handlers called on subscription 1
+        for (const id of partitionIds) {
+          partitionHandlerCalls[id] = { initialize: 0, processEvents: 0, close: 0 };
+        }
+
+        const subscriptionHandlers1: SubscriptionEventHandlers = {
+          async processError() {},
+          async processEvents(_, context) {
+            partitionHandlerCalls[context.partitionId].processEvents++;
+          },
+          async processClose(_, context) {
+            partitionHandlerCalls[context.partitionId].close++;
+            // reset processEvents count
+            partitionHandlerCalls[context.partitionId].processEvents = 0;
```

Comment:
So this is more of a boolean (we have received any events, or haven't received any), basically?

Reply:
Yes, and I can convert this to a boolean to be clearer.